tests/performance/performance.cpp

```diff
@@ -266,6 +353,124 @@ _ebpf_program_invoke()
     _ebpf_program_test_state_instance->test(nullptr);
 }
 
+static void
+_map_find_read_test(uint32_t cpu_id)
+{
+    _ebpf_map_test_state_instance->test_find_read(cpu_id);
+}
+
+static void
+_map_find_write_test(uint32_t cpu_id)
+{
+    _ebpf_map_test_state_instance->test_find_write(cpu_id);
+}
+
+static void
+_map_update_test(uint32_t cpu_id)
+{
+    _ebpf_map_test_state_instance->test_update(cpu_id);
+}
+
+static const char*
+_ebpf_map_type_t_to_string(ebpf_map_type_t type)
+{
+    switch (type) {
+    case BPF_MAP_TYPE_UNSPECIFIED:
+        return "BPF_MAP_TYPE_UNSPECIFIED";
+    case BPF_MAP_TYPE_HASH:
+        return "BPF_MAP_TYPE_HASH";
+    case BPF_MAP_TYPE_ARRAY:
+        return "BPF_MAP_TYPE_ARRAY";
+    case BPF_MAP_TYPE_PROG_ARRAY:
+        return "BPF_MAP_TYPE_PROG_ARRAY";
+    case BPF_MAP_TYPE_PERCPU_HASH:
+        return "BPF_MAP_TYPE_PERCPU_HASH";
+    case BPF_MAP_TYPE_PERCPU_ARRAY:
+        return "BPF_MAP_TYPE_PERCPU_ARRAY";
+    case BPF_MAP_TYPE_HASH_OF_MAPS:
+        return "BPF_MAP_TYPE_HASH_OF_MAPS";
+    case BPF_MAP_TYPE_ARRAY_OF_MAPS:
+        return "BPF_MAP_TYPE_ARRAY_OF_MAPS";
+    default:
+        return "Error";
+    }
+}
+
+template <ebpf_map_type_t map_type>
+void
+test_bpf_map_lookup_elem_read(bool preemptible)
+{
+    size_t iterations = PERFORMANCE_MEASURE_ITERATION_COUNT;
+    ebpf_map_test_state_t map_test_state(map_type);
+    _ebpf_map_test_state_instance = &map_test_state;
+    std::string name = __FUNCTION__;
+    name += "<";
+    name += _ebpf_map_type_t_to_string(map_type);
+    name += ">";
+    _performance_measure measure(name.c_str(), preemptible, _map_find_read_test, iterations);
+    measure.run_test();
+}
+
+template <ebpf_map_type_t map_type>
+void
+test_bpf_map_lookup_elem_write(bool preemptible)
+{
+    size_t iterations = PERFORMANCE_MEASURE_ITERATION_COUNT;
```

Comment:
Do we need to call `ebpf_core_initiate()` in this test?

Reply:
Yes, and it currently does via the _test_helper_end_to_end constructor/destructor.